src/test/LockedWithdrawal.t.sol

```diff
@@ -0,0 +1,148 @@
+        vm.warp(1_641_070_800);
+        vm.startPrank(address(vault));
+        withdrawalEscrow.registerWithdrawalRequest(alice, 1000);
+
+        // resolve pending
+        withdrawalEscrow.resolveDebtShares(1000);
+
+        vm.warp(1_641_070_900);
+
+        assertEq(withdrawalEscrow.canWithdraw(alice), true);
+
+        assertEq(withdrawalEscrow.withdrawableAmount(alice), 0);
+
+        changePrank(alice);
+        assertEq(withdrawalEscrow.redeem(), 0);
+        // check balance of alice
+        assertEq(withdrawalEscrow.balanceOf(alice), 0);
+    }
+
+    function testRedeemDealFunds() public {
+        // do a withdrawal request
+        vm.warp(1_641_070_800);
+        vm.startPrank(address(vault));
+        withdrawalEscrow.registerWithdrawalRequest(alice, 1000);
+
+        // resolve pending
+        withdrawalEscrow.resolveDebtShares(1000);
+
+        vm.warp(1_641_070_900);
+
+        assertEq(withdrawalEscrow.canWithdraw(alice), true);
+
+        // send asset to escrow
+        deal(vault.asset(), address(withdrawalEscrow), 1234);
+
+        // check alice get the full amount
+        assertEq(withdrawalEscrow.withdrawableAmount(alice), 1234);
+
+        changePrank(alice);
+        assertEq(withdrawalEscrow.redeem(), 1234);
+        // check balance of alice
+        assertEq(withdrawalEscrow.balanceOf(alice), 0);
+    }
+
+    function testSharedWithdawalAmount() public {
+        vm.warp(1_641_070_800);
+        vm.startPrank(address(vault));
+        // register alice
+        withdrawalEscrow.registerWithdrawalRequest(alice, 1000);
+
+        // register bob
+        withdrawalEscrow.registerWithdrawalRequest(bob, 2000);
+
+        withdrawalEscrow.resolveDebtShares(3000);
+
+        // allocate funds for escrow
+        deal(vault.asset(), address(withdrawalEscrow), 3000);
+
+        vm.warp(1_641_080_800);
+
+        assertEq(withdrawalEscrow.canWithdraw(alice), true);
+        assertEq(withdrawalEscrow.withdrawableAmount(alice), 1000);
+
+        changePrank(alice);
+        assertEq(withdrawalEscrow.redeem(), 1000);
+
+        assertEq(withdrawalEscrow.canWithdraw(bob), true);
+        assertEq(withdrawalEscrow.withdrawableAmount(bob), 2000);
+        changePrank(bob);
+        assertEq(withdrawalEscrow.redeem(), 2000);
+    }
+
+    function testPartialDebtClearance() public {
+        vm.warp(1_641_070_800);
+        vm.startPrank(address(vault));
+        // register alice
+        withdrawalEscrow.registerWithdrawalRequest(alice, 1000);
+
+        // register bob
+        withdrawalEscrow.registerWithdrawalRequest(bob, 2000);
+
+        withdrawalEscrow.resolveDebtShares(1000);
+
+        // allocate funds for escrow
+        deal(vault.asset(), address(withdrawalEscrow), 1000);
+
+        vm.warp(1_641_080_800);
+
+        assertEq(withdrawalEscrow.canWithdraw(alice), true);
+        assertEq(withdrawalEscrow.withdrawableAmount(alice), 1000);
+
+        assertEq(withdrawalEscrow.canWithdraw(bob), false);
```

Comment:
We should probably discuss allowing partial withdrawals.

 If bob registers a request for 2000 shares before alice and then 1000 shares are resolved, then only alice can redeem.